jupyterhub/handlers/pages.py

```diff
@@ -24,6 +27,11 @@ class RootHandler(BaseHandler):
     Otherwise, renders login page.
     """
     def get(self):
+        next_url = self.get_argument('next', '')
+        if next_url and next_url.startswith('/user/'):
```

Comment:
@minrk, I tested replacing `startswith('/user/')` by `startswith(url_path_join(self.base_url, 'user') + '/')` in @daradib's patch, and it works correctly if I don't customize the base URL.

However, if I set:

    c.JupyterHub.base_url = '/base/'

and leave this as it as:

    c.JupyterHub.hub_prefix = '/hub/'

and log in as `receiver` and try to open the link `/base/user/sender/notebooks/foo.ipynb` while sender of the link is logged in, I'm redirected to `/base/hub/base/user/sender/notebooks/foo.ipynb` (which doesn't work).

I also tried setting

    c.JupyterHub.hub_prefix = '/base/hub/'

But it made no difference.